code/game/turfs/closed/wall/window_frames.dm

```diff
@@ -39,11 +39,16 @@
 	. = ..()
 
 	update_appearance()
-	AddElement(/datum/element/climbable)
+	AddElement(/datum/element/climbable, on_try_climb_procpath = TYPE_PROC_REF(/obj/structure/window_frame/, on_try_climb))
 
 	if(mapload && start_with_window)
 		create_structure_window(window_type, TRUE)
 
+	var/static/list/loc_connections = list(
+		COMSIG_ATOM_ENTERED = PROC_REF(on_entered),
+	)
+	AddElement(/datum/element/connect_loc, loc_connections)
```

Comment:
Could you test if this means throwing a player over a low wall would shock them as they pass over? Ideally if they're being thrown over and flying like that they wouldn't be shocked

Reply:
it  probably will! good point :) I'll check for the flags

Reply:
Apparently I already do this :P 

```/obj/structure/window_frame/proc/on_entered(datum/source, AM as mob|obj)
	SIGNAL_HANDLER
	if(!isliving(AM))
		return
	var/mob/living/potential_victim = AM
	if(potential_victim.movement_type & (FLOATING|FLYING))
		return
	shock(potential_victim, 100)```